include/PathUtilities.h

```diff
@@ -0,0 +1,289 @@
+    /// @brief Get the filename portion of a string.
+    /// @param PathAndFile A complete path and filename.
+    /// @return If passed "/a/b/c.txt" or "c:\windirs\crash.exe" this will return "c.txt" or "crash.exe".
+    String MEZZ_LIB GetBaseName(const char* PathAndFile);
+    /// @brief Get the filename portion of a string.
+    /// @param PathAndFile A complete path and filename.
+    /// @return If passed "/a/b/c.txt" or "c:\windirs\crash.exe" this will return "c.txt" or "crash.exe".
+    String MEZZ_LIB GetBaseName(const StringView PathAndFile);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Separators
+
+    /// @brief Checks to see if a character is used by the host OS to separate directories.
+    /// @param ToCheck The character to check.
+    /// @return Returns true if the character is '\\' on windows, or '/' on linux, false otherwise.
+    template<typename CharType, typename = std::enable_if_t< StringTools::is_char<CharType>::value > >
+    constexpr Boole IsDirectorySeparator_Host(const CharType ToCheck)
+    {
+    #ifdef MEZZ_WINDOWS
+        return ( ToCheck == '\\' );
+    #else
+        return ( ToCheck == '/' );
+    #endif
+    }
+    /// @brief Checks to see if a character is used by posix to separate directories.
+    /// @param ToCheck The character to check.
+    /// @return Returns true if the character specified is '/'.
+    template<typename CharType, typename = std::enable_if_t< StringTools::is_char<CharType>::value > >
+    constexpr Boole IsDirectorySeparator_Posix(const CharType ToCheck)
+        { return ( ToCheck == '/' ); }
+    /// @brief Checks to see if a character is used by windows to separate directories.
+    /// @param ToCheck The character to check.
+    /// @return Returns true if the character specified is '\\'.
+    template<typename CharType, typename = std::enable_if_t< StringTools::is_char<CharType>::value > >
+    constexpr Boole IsDirectorySeparator_Windows(const CharType ToCheck)
+        { return ( ToCheck == '\\' ); }
+    /// @brief Checks to see if a character is used to separate directories.
+    /// @param ToCheck The character to check.
+    /// @return Returns true if the character specified is '\\' or '/', false otherwise.
+    template<typename CharType, typename = std::enable_if_t< StringTools::is_char<CharType>::value > >
+    constexpr Boole IsDirectorySeparator(const CharType ToCheck)
+        { return ( IsDirectorySeparator_Posix(ToCheck) || IsDirectorySeparator_Windows(ToCheck) ); }
+    /// @brief Gets the character used by the client OS to separate directories.
+    /// @return Returns a backslash '\\' on windows and forward slash '/' on other operating systems.
+    template<typename CharType = char, typename = std::enable_if_t< StringTools::is_char<CharType>::value > >
+    constexpr CharType GetDirectorySeparator_Host()
+    {
+    #ifdef MEZZ_WINDOWS
+        return '\\';
```

Comment:
To reduce duplication shouldn't these be defined in terms of GetDirectorySeparator_Posix and GetDirectorySeparator_Windows?

Ideally there would only be one '\\' and '/' and everything would be defined in terms of that. Then the functions would just focus on adding the semantic meaning they provide.

Reply:
Updated.